### PR TITLE
Chartboost - support for SDK 8.0.1

### PR DESCRIPTION
**Pull request by daniel-barros**
Added support for the new Chartboost SDK version 8.0.0 for iOS.
It includes banner ads support and some fixes.

### CODE REVIEW

File: Chartboost/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 ## Changelog
+  * 8.0.0.0
+      * This version of the adapters has been certified with Chartboost 8.0.0.
```
Comment (chauduyphanvu) [inline Chartboost/CHANGELOG.md]: I see that 8.0.1 has been released for iOS. Was there any API change that'd affect the code on this PR?

File: Chartboost/ChartboostAdapterConfiguration.m [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #import "MPLogging.h"
 #endif
 
-#define CHARTBOOST_ADAPTER_VERSION             @"7.5.0.0"
+#define CHARTBOOST_ADAPTER_VERSION             @"8.0.0.0"
```
Comment (chauduyphanvu) [inline Chartboost/ChartboostAdapterConfiguration.m]: Same as above (`8.0.1.0`)

File: Chartboost/ChartboostBannerCustomEvent.m [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+@interface ChartboostBannerCustomEvent () <CHBBannerDelegate>
+
+@property (nonatomic) CHBBanner *banner;
+@property (nonatomic, copy) NSString *appID;
+
+@end
+
+@implementation ChartboostBannerCustomEvent
+
+- (void)requestAdWithSize:(CGSize)size customEventInfo:(NSDictionary *)info
```
Comment (chauduyphanvu) [inline Chartboost/ChartboostBannerCustomEvent.m]: Please use `- (void)requestAdWithSize:(CGSize)size customEventInfo:(NSDictionary *)info adMarkup:(NSString *)adMarkup` as the current method is going to be deprecated.

File: Chartboost/CHANGELOG.md (hunk 1 above)
Comment (daniel-barros) [reply Chartboost/CHANGELOG.md]: No, 8.0.1 is just some internal fixes. I updated the version number

File: Chartboost/ChartboostAdapterConfiguration.m (hunk 2 above)
Comment (daniel-barros) [reply Chartboost/ChartboostAdapterConfiguration.m]: Done

File: Chartboost/ChartboostBannerCustomEvent.m (hunk 3 above)
Comment (daniel-barros) [reply Chartboost/ChartboostBannerCustomEvent.m]: Sure, done

**Comment by daniel-barros** [comment]
Hi @chauduyphanvu, thanks for taking the time to review this PR!
I responded to your comments and made the requested changes. Please let me know if I can do anything else to move this forward. Cheers

**Comment by daniel-barros** [comment]
Hi @chauduyphanvu, please let me know if there is anything we can do to move this PR forward.

**Comment by rsatrasala** [comment]
@daniel-barros I was working with Carmine from your team to ensure our recent banner changes are compatible with your SDK. We got the confirmation this week and we plan to start the certification next week.

**Comment by chauduyphanvu** [comment]
@daniel-barros, I'll take another look at the code, but right now we're waiting for frontend and backend support before we can do an end-to-end test and merge this PR. Stay tuned!

**Comment by daniel-barros** [comment]
Thanks @rsatrasala and @chauduyphanvu for the time you put in the review and the given feedback. We are excited to have this adapters merged and getting banner support!

File: Chartboost/ChartboostBannerCustomEvent.m [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+@implementation ChartboostBannerCustomEvent
+
+- (void)requestAdWithSize:(CGSize)size customEventInfo:(NSDictionary *)info adMarkup:(NSString *)adMarkup
+{
+    self.appID = [info objectForKey:@"appId"];
+    NSString *appSignature = [info objectForKey:@"appSignature"];
+    MPLogAdEvent([MPLogEvent adLoadAttemptForAdapter:NSStringFromClass(self.class) dspCreativeId:nil dspName:nil], self.appID);
+    
+    if ([self.appID length] == 0 || [appSignature length] == 0) {
+        NSError *error = [NSError errorWithCode:MOPUBErrorAdapterInvalid localizedDescription:@"Failed to load Chartboost banner: missing either appId or appSignature"];
```
Comment (chauduyphanvu) [inline Chartboost/ChartboostBannerCustomEvent.m]: Please add a resolution step:

```suggestion
        NSError *error = [NSError errorWithCode:MOPUBErrorAdapterInvalid localizedDescription:@"Failed to load Chartboost banner: missing either appId or appSignature. Make sure you have a valid appId or appSignature entered on the MoPub dashboard."];
```

File: Chartboost/ChartboostBannerCustomEvent.m [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], self.appID);
+        [self.delegate bannerCustomEvent:self didFailToLoadAdWithError:error];
```
Comment (chauduyphanvu) [inline Chartboost/ChartboostBannerCustomEvent.m]: Add a new line before a `return` statement.

File: Chartboost/ChartboostBannerCustomEvent.m [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        return;
+    }
+    
+    NSString *location = [info objectForKey:@"location"];
+    location = [location length] != 0 ? location: CBLocationDefault;
+    if (self.banner && (!CGSizeEqualToSize(self.banner.size, size) || ![self.banner.location isEqualToString:location])) {
+        NSError *error = [NSError errorWithCode:MOPUBErrorAdapterInvalid localizedDescription:@"Chartboost adapter failed to load ad: size or location in new request doesn't match the existing banner values."];
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], self.appID);
+        [self.delegate bannerCustomEvent:self didFailToLoadAdWithError:error];
+        return;
```
Comment (chauduyphanvu) [inline Chartboost/ChartboostBannerCustomEvent.m]: New line before `return`, here and elsewhere.

File: Chartboost/ChartboostBannerCustomEvent.m [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    
+    if ([self.appID length] == 0 || [appSignature length] == 0) {
+        NSError *error = [NSError errorWithCode:MOPUBErrorAdapterInvalid localizedDescription:@"Failed to load Chartboost banner: missing either appId or appSignature"];
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], self.appID);
+        [self.delegate bannerCustomEvent:self didFailToLoadAdWithError:error];
+        return;
+    }
+    
+    NSString *location = [info objectForKey:@"location"];
+    location = [location length] != 0 ? location: CBLocationDefault;
```
Comment (chauduyphanvu) [inline Chartboost/ChartboostBannerCustomEvent.m]: Add new line below for readability.

File: Chartboost/ChartboostBannerCustomEvent.m (hunk 4 above)
Comment (daniel-barros) [reply Chartboost/ChartboostBannerCustomEvent.m]: 👍

File: Chartboost/ChartboostBannerCustomEvent.m (hunk 5 above)
Comment (daniel-barros) [reply Chartboost/ChartboostBannerCustomEvent.m]: 👍

File: Chartboost/ChartboostBannerCustomEvent.m (hunk 6 above)
Comment (daniel-barros) [reply Chartboost/ChartboostBannerCustomEvent.m]: 👍

File: Chartboost/ChartboostBannerCustomEvent.m (hunk 7 above)
Comment (daniel-barros) [reply Chartboost/ChartboostBannerCustomEvent.m]: I moved this a few lines below so it's closer to where it's used

**Comment by daniel-barros** [comment]
> @daniel-barros, thanks for submitting the PR! I suggested some minor syntactical changes. Once again, could you confirm that Chartboost is OK with MoPub (via the adapter) sending any requested banner size to you? Here's the line:
> 
> ```objc
> if (self.banner && (!CGSizeEqualToSize(self.banner.size, size) || ![self.banner.location isEqualToString:location])) {
> ```

Thanks @chauduyphanvu. I applied the requested changes.
About the banner size, it is fine. Actually that check was a defensive measure to prevent multiple incompatible requests done on the same event. I simplified it to just prevent reusing an event.

**Pull request merged**